go/vt/vtgate/evalengine/fn_json.go

```diff
@@ -0,0 +1,366 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package evalengine
+
+import (
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/sqltypes"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/evalengine/internal/json"
+)
+
+type (
+	builtinJSONExtract struct {
+		CallExpr
+	}
+
+	builtinJSONUnquote struct {
+		CallExpr
+	}
+
+	builtinJSONObject struct {
+		CallExpr
+	}
+
+	builtinJSONArray struct {
+		CallExpr
+	}
+
+	builtinJSONDepth struct {
+		CallExpr
+	}
+
+	builtinJSONLength struct {
+		CallExpr
+	}
+
+	builtinJSONContainsPath struct {
+		CallExpr
+	}
+
+	builtinJSONKeys struct {
+		CallExpr
+	}
+)
+
+var _ Expr = (*builtinJSONExtract)(nil)
+var _ Expr = (*builtinJSONUnquote)(nil)
+var _ Expr = (*builtinJSONObject)(nil)
+var _ Expr = (*builtinJSONArray)(nil)
+var _ Expr = (*builtinJSONDepth)(nil)
+var _ Expr = (*builtinJSONLength)(nil)
+var _ Expr = (*builtinJSONContainsPath)(nil)
+var _ Expr = (*builtinJSONKeys)(nil)
+
+var errInvalidPathForTransform = vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "In this situation, path expressions may not contain the * and ** tokens or an array range.")
```

Comment:
nitpick: I think the recommendation for error strings is to not be capitalized or end with punctuation